### ISSUE TITLE
Clarification on adding classNames to Overlay component

### ISSUE THREAD

**Issue by nselikoff** [sandydoo/ember-google-maps#89]
### Describe the bug
I just tried adding a class to an overlay component with the `classNames` attribute per the [README](https://github.com/sandydoo/ember-google-maps#examples-of-what-to-expect), but the class isn't inserted into the overlay component in the DOM.

### To Reproduce
```hbs
{{#g-map lat='51.508530' lng='-0.076132' zoom=10 as |g|}}
  {{#each rentals as |r|}}
    {{#g.overlay lat=r.lat lng=r.lng classNames='custom-marker'}}
      <div class="marker-content">
        <p class="price">{{r.price}}</p>
      </div>
    {{/g.overlay}}
  {{/each}}
{{/g-map}}
```

Check DOM, `custom-marker` class is not added to the overlay div.

### Expected behaviour
The overlay div has the `custom-marker` class.

### Attempted solutions
Reviewing the code a bit, it seems adding custom class names is only supported on the Map component. Is the intention to be able to do this with Overlay as well? It would help me target certain overlays for CSS transitions on the `transform` property.

--------------------------------------------------------------------------------

### Addon configuration and device details

**Addon configuration**:

 - Addon version: 3.0.0-beta.2
 
 - Addon config from `environment.js` (**do not** post your API key): 
```js
    "ember-google-maps": {
      key: process.env.GOOGLE_MAPS_API_KEY,
      protocol: "https",
      language: "en",
      version: "3.38",
      libraries: ["visualization", "places"]
    },
```
 
 - Google Maps version: 3.38

**Comment by sandydoo** [comment]
Hi @nselikoff,

> adding a class to an overlay component with the classNames attribute per the README

You're right, that won't work. I'll fix the README.

> It would help me target certain overlays for CSS transitions on the transform property.

This actually wouldn't work out that well.

The way these overlays work is that Google Maps calls the `draw` function using `requestAnimationFrame` whenever it needs to update the position of the overlay.

On each draw, we convert the coordinates to pixels offsets and manually set the style with the calculated transform (going via Glimmer/Ember is not an option because of that animation frame).

What happens if you set a class with a transform on this element? Well, it's just going to be overwritten pretty much instantly. If you set `transition`, the element will stagger whenever you try to zoom the map. Also, in general, these elements are not a good target for animation. They end up being transformed to something like `translateX(30px) translateY(-71px)`, so animating to and from these values reliably is a problem.

What does work is something like this (in Octane syntax):

```hbs
<g.overlay
  @lat={{l.lat}}
  @lng={{l.lng}}
  @zIndex={{if l.active 1000 index}}
  @onMouseover={{action (mut l.active) }}
  @onMouseleave={{action (mut l.active) false}}>
  <div class="animate-drop-down">
    <div class="anchor-middle-bottom">
      <div class="tooltip {{if l.active "active"}}">
        £{{l.price}}
      </div>
    </div>
  </div>
</g.overlay>
```

```css
.animate-drop-down {
  animation: slide-down 0.4s ease;
}

@keyframes slide-down {
  0% {
    opacity: 0;
    transform: translateY(-20px);
  }
  100% {
    opacity: 1;
    transform: translateY(0);
  }
}
```

Note that you have full control over your element and you are working in a zeroed-out pixel coordinate space.

To recap, that inner element is strictly for positioning the overlay on the map. All styling should be applied to the nested elements. The only exception is the `z-index`, which can't be set properly when you nest divs, so that got its own custom argument.

I'll try to find some time to write this out in the docs and push out the 3.0 docs as well.

Hope this clarifies things a bit. Let me know if there are any issues with this!

**Comment by nselikoff** [comment]
Hi @sandydoo,

Thanks for the thorough explanation, that all makes sense. 

In my case, I've got overlays representing vehicles on a map whose positions get updated about every 30 seconds. Instead of jumping to the new position I was hoping to take advantage of CSS transitions to smoothly animate to the new position. I hadn't thought about the side effect of staggering while zooming/panning if a transition was taking place, but that makes perfect sense given what's happening in the implementation. 

Ultimately to implement this I'd have to drop down a level and interpolate between the old and new lat/lng over the course of some amount of time. And realistically at that point the elements should probably not be Ember components as the overhead of changing a watched property would likely make it difficult to achieve a smooth 30-60fps transition.

For another day, then. Thanks!
